tests/Microsoft.Identity.Test.Unit.net45/AppConfigTests/PublicClientApplicationBuilderTests.cs

```diff
@@ -0,0 +1,171 @@
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+// 
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+// 
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+// 
+// ------------------------------------------------------------------------------
+
+using System.Net.Http;
+using Microsoft.Identity.Client;
+using Microsoft.Identity.Client.AppConfig;
+using Microsoft.Identity.Client.Core;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Microsoft.Identity.Test.Unit.AppConfigTests
+{
+    [TestClass]
+    [TestCategory("BuilderTests")]
+    public class PublicClientApplicationBuilderTests
+    {
+        [TestMethod]
+        public void TestConstructor()
+        {
+            var pca = PublicClientApplicationBuilder.Create(MsalTestConstants.ClientId).Build();
+            Assert.AreEqual(MsalTestConstants.ClientId, pca.ClientId);
+            Assert.IsNotNull(pca.UserTokenCache);
+
+            // Validate Defaults
+            Assert.AreEqual(LogLevel.Warning, pca.AppConfig.LogLevel);
+            Assert.IsNull(pca.AppConfig.ClientCredential);
+            Assert.AreEqual(MsalTestConstants.ClientId, pca.AppConfig.ClientId);
+            Assert.IsNull(pca.AppConfig.Component);
+            Assert.AreEqual(false, pca.AppConfig.EnablePiiLogging);
+            Assert.IsNull(pca.AppConfig.HttpClientFactory);
+            Assert.AreEqual(false, pca.AppConfig.IsDefaultPlatformLoggingEnabled);
+            Assert.IsNull(pca.AppConfig.LoggingCallback);
+            Assert.AreEqual(Constants.DefaultRedirectUri, pca.AppConfig.RedirectUri);
+            Assert.IsNull(pca.AppConfig.TelemetryCallback);
+            Assert.AreEqual(null, pca.AppConfig.TenantId);
+        }
+
+        [TestMethod]
+        public void TestWithDifferentClientId()
+        {
+            var pca = PublicClientApplicationBuilder.Create("this is a test client id").Build();
+            Assert.AreEqual("this is a test client id", pca.ClientId);
+        }
+
+        [TestMethod]
+        public void TestConstructor_ClientIdOverride()
+        {
+            var pca = PublicClientApplicationBuilder.Create(MsalTestConstants.ClientId).WithClientId("some other client id").Build();
```

Comment:
>Create(MsalTestConstants.ClientId).WithClientId("some other client id") [](start = 53, length = 71)

Seems odd. given ClientID is part of the Create, is this even needed? Or perhaps we can remove it from Create?

Reply:
WithClientId is needed if you load from options and don't have the client id specified there, you still need a way to add it or override it.  this is just testing the behavior if you call it multiple times.